File: NextApp/iOS/OrgPortalNext/Core/DataLayer/MeetingMinutesRepository.swift
import Foundation
import Model

@MainActor
public protocol MeetingMinutesRepository {
    func fetchAll() async throws -> [MeetingMinutes]
    func save(_ minutes: MeetingMinutes) async throws
    func delete(id: UUID) async throws
}

public protocol MeetingRecordingDraftStoring: Sendable {
    func load() throws -> MeetingRecordingDraft?
    func save(_ draft: MeetingRecordingDraft) throws
    func delete() throws
}

File: NextApp/Android/core/data/src/main/kotlin/jp/komehyappyo/member/next/core/data/MeetingMinutesRepository.kt
package jp.komehyappyo.member.next.core.data

import jp.komehyappyo.member.next.core.model.MeetingMinutes
import jp.komehyappyo.member.next.core.model.MeetingRecordingDraft
import kotlinx.coroutines.flow.Flow
import java.util.UUID

interface MeetingMinutesRepository {
    fun observeAll(): Flow<List<MeetingMinutes>>
    suspend fun save(minutes: MeetingMinutes)
    suspend fun delete(id: UUID)
}

interface MeetingRecordingDraftStore {
    fun load(): MeetingRecordingDraft?
    fun save(draft: MeetingRecordingDraft)
    fun delete()
}
